### PR TITLE
Port Fixes and enhancements to legacy builder

### LINKED ISSUE #127: Back fix the logic fixes for reorder test refactor to legacy
The logic fixes contained with #114 still exist in the legacy branch of the builder.  Based on conversations with @jbearer we do intend to run the `legacy` builder for a while still, so we should backport the logic fixes for cloning the `BuilderState`s to the legacy builder as well.

### PR DESCRIPTION
**Pull request by Ayiga**
Adjust `service.rs` so that specific `BuilderState` information can be tracked

Apply changes to address unecessary `BuilderState` clones

Closes #127

### This PR:

Ports over the fixes and updates applied to the Marketplace builder to the legacy builder.  Since these both use the same logic, the same improvements of behavior and logic can be applied to the legacy.

### CODE REVIEW

File: crates/legacy/src/lib.rs [hunk 1]
```diff
@@ -78,3 +80,18 @@ impl<Types: NodeType> std::fmt::Display for BuilderStateId<Types> {
         write!(f, "BuilderState({}@{})", self.parent_commitment, *self.view)
     }
 }
+
+/// References to the parent block that is extended to spawn the new builder state.
+#[derive(Debug, Clone)]
+pub struct ParentBlockReferences<TYPES: NodeType> {
```
Comment (shenkeyao) [inline crates/legacy/src/lib.rs:86]: Looks like the same struct in `utils.rs` can be used by both legacy and marketplace code so we can remove it here.

**Review by shenkeyao — COMMENTED** [review]
A minor comment about potentially removing the duplicated `ParentBlockReferences`.

File: crates/legacy/src/lib.rs (hunk 1 above)
Comment (Ayiga) [reply crates/legacy/src/lib.rs:86]: They exist in separate `crate`s.  So do we want the legacy builder to depend on the `marketplace`?

Comment (shenkeyao) [reply crates/legacy/src/lib.rs:112]: Oh, I didn't realize `utils.rs` was in the marketplace crate. No need to change then.

**Review by shenkeyao — APPROVED** [review]

File: crates/legacy/src/lib.rs (hunk 1 above)
Comment (Ayiga) [reply crates/legacy/src/lib.rs:112]: We could eventually create a shared crate between them.  But I think we can probably do that at a later point.

Comment (shenkeyao) [reply crates/legacy/src/lib.rs:112]: Sounds good. Made an issue: https://github.com/EspressoSystems/marketplace-builder-core/issues/139.

**Pull request merged**